### PR TITLE
Fix divide by zero crash when timebase < 10000

### PR DESCRIPTION
**Pull request by elv-reza**
This doesn't have impact on normal cases when timebase >= 10000.

Tested with timebase bigger and smaller than 10000, both works fine.

### CODE REVIEW

File: libavformat/dashenc.c [hunk 1]
```diff
@@ -1739,11 +1751,19 @@ static int dash_flush(AVFormatContext *s, int final, int stream)
...
-            os->muxer_overhead = ((int64_t) (range_length - os->total_pkt_size) *
+        if (!os->muxer_overhead) {
+            if (av_rescale_q(os->max_pts - os->start_pts, st->time_base, AV_TIME_BASE_Q) != 0)
+                os->muxer_overhead = ((int64_t) (range_length - os->total_pkt_size) *
                                   8 * AV_TIME_BASE) /
                                  av_rescale_q(os->max_pts - os->start_pts,
                                               st->time_base, AV_TIME_BASE_Q);
+            else
+                /* This (else) should not happen anymore after rescaling seg_duration_ts.
+                 * But just to be in safe side I will keep this.
```
Comment (elv-serban) [inline libavformat/dashenc.c]: '... just to be on the safe side'

File: libavformat/dashenc.c [hunk 2]
```diff
@@ -1739,11 +1751,19 @@ static int dash_flush(AVFormatContext *s, int final, int stream)
             }
         }
 
-        if (!os->muxer_overhead)
-            os->muxer_overhead = ((int64_t) (range_length - os->total_pkt_size) *
+        if (!os->muxer_overhead) {
+            if (av_rescale_q(os->max_pts - os->start_pts, st->time_base, AV_TIME_BASE_Q) != 0)
```
Comment (elv-serban) [inline libavformat/dashenc.c]: What is this 'if' mean here - what condition is needed to to rescale vs no rescale?

Comment (elv-reza) [reply libavformat/dashenc.c]: You mean muxer_overhead, right? it is a parameter that is used to calculate bitrate.

Comment (elv-reza) [reply libavformat/dashenc.c]: If you mean when does av_rescale_q() rescale, I should say, most of the time it rescales to original timabase (when timebase > 10000), and in the crash case it rescaled to zero (which was wrong). It should rescale to some big number, like 2000000 when timebase < 10000.

**Pull request merged**